src/services/url.ts

```diff
@@ -72,10 +72,10 @@ class UrlService extends BaseService<UrlService.Options> {
     });
     switch (route) {
       case Routes.SEARCH:
-        this.app.flux.store.dispatch(<any>this.app.flux.actions.fetchProducts());
+        this.app.flux.awaitBiasingRehydration(this.app.flux.actions.fetchProducts());
```

Comment:
instead of making this call, can we dispatch an action (the same way the other routes are handled) and we can make a saga that's responsible for checking for hydration and doing the fetch or waiting depending on the state of the store